### PR TITLE
Lineage registration fix headless mode

### PR DESCRIPTION
**Pull request by maarzt**
Fiji with `mastodon-tomancak` installed would fail during startup when executed in headless mode.

The LineageRegistrationControlService caused this problem. It is initialized during the startup of Fiji, and it tried to create a JFrame, which fails in headless mode. And as a consequence causes Fiji to crash.

This is now fixed by the lazily initialition of the `JFrame` in the `LineageRegistrationControlService.showDialog()` method.

There is also I unit test which should detect similar problems in the future.

### CODE REVIEW

**Comment by stefanhahmann** [comment]
Hint to myself the bug could be provoked by running `ImageJ-win64.exe --headless --console`

File: src/test/java/org/mastodon/mamut/tomancak/ContextCreationTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+public class ContextCreationTest
+{
+	@Test
+	public void testCreateContext()
+	{
+		// Make sure that the SciJava context with all the Mastodon services can be created.
+		// This must work also in headless mode.
+		try (Context ignore = new Context())
+		{
+
```
Comment (stefanhahmann) [inline src/test/java/org/mastodon/mamut/tomancak/ContextCreationTest.java:57]:
```suggestion
assertNotNull( ignore );
```

It is not strictly required to add at least one assertion to a unit test, but you may consider to add testing if the context is not null.

**Review by stefanhahmann — APPROVED** [review]
The PR fixes a bug that may occur when starting Fiji with the tomancak artefact installed. The added unit test proves that also for other use cases in the future. You may consider to add an assertion to this test, but this is not strictly required.

I tested the behaviour with and without the bugfix and could reproduce both the bug and the fix. I did so running `ImageJ-win64.exe --headless --console`

File: src/test/java/org/mastodon/mamut/tomancak/ContextCreationTest.java (hunk 1 above)
Comment (maarzt) [reply src/test/java/org/mastodon/mamut/tomancak/ContextCreationTest.java]: I agree. It's really good to have this assertion there. The unit test no longer look incomplete. And the assertion makes to purpose of the test clearer.
